EZP-30806: Added example of REST search query with logical operators (#2879)
ezsystems/ezpublish-kernel: doc/specifications/rest/REST-API-V2.rst

:Error codes:
    :400: If the Input does not match the input schema definition, In this case the response contains an ErrorMessage_

Using logical operators in views
``````````

The criteria model allows combining criteria using logical operators:
- AND,
- OR,
- NOT.

By default, if multiple criteria are given, but not wrapped by any operator, an AND operator is used.

> *However* when trying to use the same type of criterion for multiple, parser will wrap it in an OR

Reviewer: ```suggestion
*However*, when trying to use the same type of criterion for multiple times, the parser will wrap it with the OR
```

operator. While this is a side effect of internal operations of the view query parser, trying to make

> an AND query for different values of the same criterion type has no practical sense (would always

Reviewer: ```suggestion
the AND query for different values of the same criterion type has no practical sense as it would always
```

> return 0 results).

Reviewer: ```suggestion
return 0 results.
```

      <ContentQuery>
        <Filter>
            <AND>
                <OR>
                    <ContentTypeIdentifierCriterion>folder</ContentTypeIdentifierCriterion>
                    <ContentTypeIdentifierCriterion>article</ContentTypeIdentifierCriterion>
                </OR>
                <SectionIdentifierCriterion>standard</SectionIdentifierCriterion>
            </AND>
        </Filter>
        <limit>10</limit>
        <offset>0</offset>
        <SortClauses>
          <ContentName>ascending</ContentName>
        </SortClauses>
      </ContentQuery>
    </ViewInput>

JSON Example
'''''''''''

The above example has the following JSON equivalent.

.. code:: json

    {
      "ViewInput": {
        "identifier": "test",
        "ContentQuery": {
          "Filter": {
            "AND": {
              "OR": {
                "ContentTypeIdentifierCriterion": [
                  "folder",
                  "article"
                ]
              },
              "SectionIdentifierCriterion": "standard"
            }
          },
          "limit": "10",
          "offset": "0",
          "SortClauses": { "ContentName": "ascending" }
        }
      }
    }

> Please notice that the structure for ContentTypeIdentifierCriterion with multiple values is slightly

Reviewer: ```suggestion
Note that the structure for ContentTypeIdentifierCriterion with multiple values is slightly
```

> different in JSON format, because parser excepts keys to be unique.

Reviewer: ```suggestion
different in JSON format, because the parser excepts keys to be unique.
```
